gax/src/main/java/com/google/api/gax/tracing/OpencensusTracerFactory.java

```diff
@@ -78,31 +82,33 @@ public OpencensusTracerFactory(@Nullable String clientNameOverride) {
    * disambiguate spans created outer manual written wrappers and around generated gapic spans.
    *
    * @param internalTracer the Opencensus tracer to wrap.
-   * @param clientNameOverride the client name that will override all of the spans' client name.
    */
   @InternalApi("Visible for testing")
-  OpencensusTracerFactory(Tracer internalTracer, @Nullable String clientNameOverride) {
+  OpencensusTracerFactory(Tracer internalTracer, @Nonnull Map<String, String> spanAttributes) {
     this.internalTracer =
         Preconditions.checkNotNull(internalTracer, "internalTracer can't be null");
-    this.clientNameOverride = clientNameOverride;
+    ImmutableMap.Builder<String, AttributeValue> formattedSpanAttributes = ImmutableMap.builder();
+    for (Map.Entry<String, String> entry : spanAttributes.entrySet()) {
+      formattedSpanAttributes.put(
+          entry.getKey(), AttributeValue.stringAttributeValue(entry.getValue()));
+    }
+    this.spanAttributes = formattedSpanAttributes.build();
   }
 
   /** {@inheritDoc } */
   @Override
   public ApiTracer newTracer(SpanName spanName) {
-    if (clientNameOverride != null) {
-      spanName = spanName.withClientName(clientNameOverride);
-    }
-    Span span = internalTracer.spanBuilder(spanName.toString()).setRecordEvents(true).startSpan();
+    SpanBuilder sb = internalTracer.spanBuilder(spanName.toString());
+    sb = sb.setRecordEvents(true);
```

Comment:
Minor: 
Looks like a non-idiomatic usage of a builder.
It would look more familiar if it was like:
```java
Span span = internalTracer.spanBuilder(spanName.toString())
        .setRecordEvents(true)
        .startSpan();
```
The way it is writtent now attracted my attention and it took some time to realize what is happening here.

Reply:
oops, that was a left over from debugging something else. Fixed